### PR TITLE
Sully gnome scrape

### PR DESCRIPTION
**Pull request by al1Null**
Started data scrape code which collects streamer info from sullygnome.com. The module writes streamer data to a .csv which can then be imported to a spreadsheet. 

Not finished, still need to collect some fields.

Let me know if you need/want help running it locally.

### CODE REVIEW

**Comment by glottsi** [comment]
I haven't really used Python so I don't feel like I can really review the code that well, maybe @chococappucino can take a look, since she knows Python?

Also, how often would we need to run this? What kind of data is more or less "permanent", and what stuff should we be trying to update? Is it possible to scrape the data once and store it in .json files so we can continue working on the site and algorithm for matching?

**Comment by al1Null** [comment]
> Also, how often would we need to run this? What kind of data is more or less "permanent", and what stuff should we be trying to update? 
Fields that I think would need at least monthly updating would be: followers, average views, and average stream length. Possibly also fields like is_partnered and average stream time may also need to be updated once in a while.

> Is it possible to scrape the data once and store it in .json files so we can continue working on the site and algorithm for matching?
Yes, after I finish scraping all the fields we need I can have it so it writes to a json file.

File: scrape/scrape.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+if __name__ == "__main__":
+
+    # headless firefox browser driver
+    options = Options()
+    options.add_argument('--headless')
+    driver = webdriver.Firefox(options=options,
+                               executable_path="geckodriver.exe")
+
+    # load twitch users
+    with open("users.json", "r") as fo:
```
[the remark this answers is not in the document]
Comment (al1Null) [reply scrape/scrape.py:72]: I'm not sure on how to go about finding new twitch users,

File: scrape/scrape.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    stats = soup.findAll(class_="MiddleSubHeaderItemValue")
+    data['partnered']       = stats[2].text == "Partnered"
+    data['language']        = stats[4].text
+    data['created']         = stats[5].text
+    data['mature']          = stats[3] == "Yes"
+    data['followers']       = int(stats[0].text.replace(",", ""))
+    data['average_viewers'] = int(soup.find(class_="InfoStatPanelTLCell").text.replace(",", ""))
+
+    ### average stream length
+    streams_url = f"https://sullygnome.com/channel/{username}/30/streams"
```
[the remark this answers is not in the document]
Comment (al1Null) [reply scrape/scrape.py:42]: Yeah, there are still some fields that have not been scraped

**Comment by glottsi** [comment]
Does this need any more work before merging? If one of you guys can merge it, or let me know what we need before merging.

**Comment by al1Null** [comment]
We still need to collect the fields
- [ ] content (as @CrashLaker suggested)
- [ ] stream frequency (n streams / week)
- [ ] average stream start time
- [ ] profile picture

File: scrape/scrape.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (chococappucino) [reply scrape/scrape.py:72]: Fixed viewer is better because we want streamers with more than 500 viewers.

**Comment by chococappucino** [comment]
Has this been collected yet for me to see?

**Comment by al1Null** [comment]
> Has this been collected yet for me to see?

ya look at `scrape/streamers2020-07-08.csv` you can import this csv file into a spreadsheet

**Pull request merged**